src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
 	 * @return scrolled page result
+	 * @deprecated since 4.0, use {@link #searchScrollStart(long, Query, Class, IndexCoordinates)}.
+	 */
+	@Deprecated
+	default <T> ScrolledPage<T> startScroll(long scrollTimeInMillis, Query query, Class<T> clazz,
+			IndexCoordinates index) {
+		return (ScrolledPage<T>) SearchHitSupport
+				.unwrapSearchHits(searchScrollStart(scrollTimeInMillis, query, clazz, index));
+	}
+
+	/**
+	 * Returns next scrolled page.
+	 *
+	 * @param scrollId the scroll id
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param clazz The class of entity to retrieve.
+	 * @return scrolled page result
+	 * @deprecated since 4.0, use {@link #searchScrollStart(long, Query, Class, IndexCoordinates)}.
 	 */
-	<T> ScrolledPage<T> startScroll(long scrollTimeInMillis, Query query, Class<T> clazz, IndexCoordinates index);
+	@Deprecated
+	default <T> ScrolledPage<T> continueScroll(@Nullable String scrollId, long scrollTimeInMillis, Class<T> clazz) {
+		return (ScrolledPage<T>) SearchHitSupport
+				.unwrapSearchHits(searchScrollContinue(scrollId, scrollTimeInMillis, clazz));
+	}
 
-	<T> ScrolledPage<T> continueScroll(@Nullable String scrollId, long scrollTimeInMillis, Class<T> clazz);
+	/**
+	 * Returns next scrolled page
+	 *
+	 * @param scrollId the scroll id
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param clazz The class of entity to retrieve.
+	 * @return scrolled page result
+	 */
+	<T> ScrolledPage<SearchHit<T>> searchScrollContinue(@Nullable String scrollId, long scrollTimeInMillis,
+			Class<T> clazz);
 
 	/**
 	 * Clears the search contexts associated with specified scroll ids.
 	 *
 	 * @param scrollId the scroll id
+	 * @deprecated since 4.0, use {@link #searchScrollClear(String)}.
 	 */
-	void clearScroll(String scrollId);
+	@Deprecated
+	default void clearScroll(String scrollId) {
+		searchScrollClear(scrollId);
+	}
 
 	/**
 	 * more like this query to search for documents that are "like" a specific document.
 	 *
+	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return page with the results
+	 * @deprecated since 4.0, use {@link #searchMoreLikeThis(MoreLikeThisQuery, Class, IndexCoordinates)}.
+	 */
+	@Deprecated
+	default <T> AggregatedPage<T> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index) {
+		return (AggregatedPage<T>) SearchHitSupport.unwrapSearchHits(searchMoreLikeThis(query, clazz, index));
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return the first returned object.
+	 *
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return the first matching object
+	 */
+	default <T> SearchHit<T> searchForObject(Query query, Class<T> clazz, IndexCoordinates index) {
```

Comment:
Let's stick with `searchOne` to align with the `findOne`/`selectOne` naming scheme that we have in the other modules.